packages/browser/src/get-selector-box-size.js

```diff
@@ -1,26 +1,153 @@
 const getSelectorBoxSize = (window, selector) => {
-  const isNotWrapperElement = (element, index, array) => {
-    const isWrapper = array.some(node =>
-      node === element ? false : element.contains(node)
-    );
-    return !isWrapper;
-  };
+  function hasOverflow(element) {
+    const style = window.getComputedStyle(element);
 
-  const isVisisble = element => {
+    if (
+      ['auto', 'hidden', 'scroll'].includes(style.overflowY) ||
+      ['auto', 'hidden', 'scroll'].includes(style.overflowX) ||
+      ['auto', 'hidden', 'scroll'].includes(style.overflow)
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  function hasFixedPosition(element) {
     const style = window.getComputedStyle(element);
+    return style.position === 'fixed';
+  }
+
+  function isElementHiddenByOverflow(
+    element,
+    { hasParentFixedPosition, hasParentOverflowHidden }
+  ) {
+    const checkOutOfBounds = () => {
+      try {
+        const elementRect = element.getBoundingClientRect();
+        const containerRect = hasParentOverflowHidden.getBoundingClientRect();
+        const top = elementRect.top < containerRect.top;
+        const bottom = elementRect.bottom > containerRect.bottom;
```

Comment:
Maybe I don't understand
But shouldn't this be "less than" to be outside the container?

```suggestion
        const bottom = elementRect.bottom < containerRect.bottom;
```

Reply:
This is correct. If the child's bottom part is greater than the container's bottom we don't want that element.

Reply:
I see 👍

Reply:
Shouldn't all the comparisons be "greater than" then? 🤔
I guess I still don't get it 🙈 

Could you please elaborate more on how this works?

Reply:
Hope this helps to explain: https://imgur.com/a/AB84p4m

Reply:
I finally get it 😅 

Got confused because of the similarities between [element positioning](https://developer.mozilla.org/en-US/docs/Web/CSS/position) and the returned value of [`getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect). Both use the same `top, right, bottom, left` names but with different meanings.

Positioning | getBoundingClientRect
------------ | -------------
![unnamed](https://user-images.githubusercontent.com/3720424/78463011-5b992c80-76d8-11ea-931b-d2517af083f2.png) | ![element-box-diagram](https://user-images.githubusercontent.com/3720424/78463057-ef6af880-76d8-11ea-9188-0a7651c7520f.png)